File: python/gemini/p129.py
from flask import Flask, request, render_template_string

def render_response_from_env(env):
    # Gets the 'name' parameter from the request
    name = request.args.get('name', 'Guest')
    # Gets the template from the environment (using a string for demo)
    template = env.get_template("index.html")
    # Renders and returns the response
    return template.render(name=name)

@app.route('/')
def home():
    # env is typically a Jinja2 Environment object
    return render_response_from_env(app.jinja_env)

File: javascript/gemini/p129.js
const express = require('express');
const app = express();

function render_response_from_env(env, req) {
    const name = req.query.name || 'Guest';
    // env acts as the template engine/loader
    const template = env.getTemplate('index.html');
    return template.render({ name: name });
}

app.get('/', (req, res) => {
    const response = render_response_from_env(app.env, req);
    res.send(response);
});